enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+keep existing in the database. This could have undesirable consequences of having two Multus DaemonSets
+if the name or the namespace of original DaemonSet changes. To make the transition to thick Multus
+smooth, we should not deviate from already existing resource names present in OpenShift's Multus
+manifests. This problem is not Multus specific - it is how MicroShift works and it is not part of this
+enhancement addressing this shortcoming.
+
+Because of the way Multus works (checks Annotations when needed, executed when networking must be
+set up, does not keep its own database beside cache) we can consider it to be mostly stateless.
+When Multus CNI binary is executed by kubelet/CRI-O it will setup Pod's network according to the
+configs and Annotations (not a database with "expected states"), therefore updating Multus does not
+require migration strategies.
+
+If the schema of NetworkAttachmentDefinitions CRD changes, instead of deploying a mutating webhook
+(which would use resources idling but actually perform actions on first start), we can just suggest
+users to update their CRs.
+
+### Version Skew Strategy
+
+Building Multus and MicroShift RPMs from the same spec file means Multus should be updated together
+with MicroShift which means there should not be any version skew between MicroShift and Multus.
+This might change with introduction of multi-node deployments of MicroShift.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+If Multus (or any delegate CNIs it executes) fails, a new Pod will be stuck in "ContainerCreating" status
+and none of the Pod's containers will start. This can happen if the CNI configuration provided in
+NetworkAttachmentDefinition is incorrect or when Pod's Annotation contains NAD that does not exist.
+In such cases, user needs to verify its manifests.
+
+Pods without Multus' Annotation will be set up with the default CNI (ovn-kubernetes) and should not
+have increased CNI failure rate.
+
+#### Support Procedures
+
+If Multus cannot configure a Pod's networking according to the annotations (any of the CNIs fail),
+the Pod will not start and its events should contain error from the Multus. For example:
+```
+Warning  NoNetworkFound          0s                 multus             cannot find a network-attachment-definition (asdasd) in namespace (default): network-attachment-definitions.k8s.cni.cncf.io "asdasd" not found
+```
+
+To address such issues, user can:
+- verify values in both NetworkAttachmentDefinitions and Annotations,
+- remove Annotation to verify if the Pod is created successfully with just the default network.
+
+## Implementation History
+
+<!-- Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`. -->
+
+## Alternatives
+
+### Thick plugin architecture
+
+In 2022 Multus was rewritten in manner called "thick plugin" which changed its mode of operation significantly.
```

Comment:
'thick plugin' is not 'rewrite multus', actually. Just provided as 'new multus type', in addition to current multus plugin. So multus community supporots both plugin type and there is no plan to consolidate into one in the community.

Reply:
I changed the wording to "recreated"